README.md

```diff
@@ -20,3 +20,25 @@ If you have any question or idea regarding the project, please feel free to reac
 ## License
 
 **hermes** is published under [Apache License 2.0](http://www.apache.org/licenses/LICENSE-2.0).
+
+## Development 
+
+### Code formatting
+For code formatting we use [google-java-format](https://github.com/google/google-java-format/tree/master).
+Following steps are required for optimal dev experience in IJ:
+
+1. Download [google-java-format plugin](https://plugins.jetbrains.com/plugin/8527-google-java-format)
+2. [Set custom VM options required for IJ plugin](https://github.com/google/google-java-format/tree/master?tab=readme-ov-file#intellij-jre-config)
+3. Go to `Settings > google-java-format` and click `Enable google java-format` (should be checked by default)
+4. Go to `Settings > Tools > Actions on Save` and enable `Reformat code` and `Optimize imports` for Java files
+
+Each save should automatically trigger reformat.
+
+If you want to debug the CLI check:
+
+```shell
+wget https://github.com/google/google-java-format/releases/download/v1.23.0/google-java-format_darwin-arm64
```

Comment:
**note:** Here we implicitly assume that everyone is using Apple Silicon Macintosh